test_importlib fails when _testcapi is a builtin module

**Bug report**

Some test_importlib cases for extension modules are failing when Python is compiled with dynamic extension support but ``_testcapi`` or ``_testmultiphase`` are builtin modules:

```
======================================================================
FAIL: test_module (test.test_importlib.extension.test_finder.Frozen_FinderTests.test_module)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/python-wasm/cpython/Lib/test/test_importlib/extension/test_finder.py", line 25, in test_module
    self.assertTrue(self.find_spec(util.EXTENSIONS.name))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: None is not true
```

The tests should be skipped if the module name is in ``sys.builtin_module_names``.

--- a/Lib/test/test_importlib/extension/test_finder.py
+++ b/Lib/test/test_importlib/extension/test_finder.py
@@ -3,7 +3,7 @@ from test.test_importlib import abc, util
 machinery = util.import_importlib('importlib.machinery')
 
 import unittest
-import warnings
+import sys
 
 
 class FinderTests(abc.FinderTests):
@@ -13,6 +13,10 @@ class FinderTests(abc.FinderTests):
     def setUp(self):
         if not self.machinery.EXTENSION_SUFFIXES:
             raise unittest.SkipTest("Requires dynamic loading support.")
+        if util.EXTENSIONS.name in sys.builtin_module_names:
+            raise unittest.SkipTest(
+                f"{util.EXTENSIONS.name} is a builtin module"
+            )
 
     def find_spec(self, fullname):
         importer = self.machinery.FileFinder(util.EXTENSIONS.path,

--- a/Lib/test/test_importlib/extension/test_loader.py
+++ b/Lib/test/test_importlib/extension/test_loader.py
@@ -20,6 +20,10 @@ class LoaderTests(abc.LoaderTests):
     def setUp(self):
         if not self.machinery.EXTENSION_SUFFIXES:
             raise unittest.SkipTest("Requires dynamic loading support.")
+        if util.EXTENSIONS.name in sys.builtin_module_names:
+            raise unittest.SkipTest(
+                f"{util.EXTENSIONS.name} is a builtin module"
+            )
         self.loader = self.machinery.ExtensionFileLoader(util.EXTENSIONS.name,
                                                          util.EXTENSIONS.file_path)
 
@@ -97,6 +101,10 @@ class MultiPhaseExtensionModuleTests(abc.LoaderTests):
         if not self.machinery.EXTENSION_SUFFIXES:
             raise unittest.SkipTest("Requires dynamic loading support.")
         self.name = '_testmultiphase'
+        if self.name in sys.builtin_module_names:
+            raise unittest.SkipTest(
+                f"{self.name} is a builtin module"
+            )
         finder = self.machinery.FileFinder(None)
         self.spec = importlib.util.find_spec(self.name)
         assert self.spec

